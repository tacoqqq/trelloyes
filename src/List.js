import React from 'react';
import './List.css';
import Card from './Card.js';




function List(props){
    console.log(props);
    console.log(props.cards);
    const cardArray = (Array.isArray(props.cards)) ? props.cards.map(card => <Card key={card.id} title={card.title} content={card.content}/>) : ""; // [{title:'xxx',content:'yyy'},{title:'xxx',content:'yyy'}]

    //const cardComponentArray = (Array.isArray(cardArray) || !cardArray == {}) ? 
    
    return (
        <section className='List'>
            <header className='List-header'>
                <h2>{props.header}</h2>
            </header>
            <div className='List-cards'>
                {cardArray}
            </div>
        </section>
    );
}

export default List;