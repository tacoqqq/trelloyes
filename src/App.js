import React from 'react';
import List from './List.js';
import './App.css';


function App(props) {
  const allList = props.store.lists ? props.store.lists:[]
  //console.log(allList);
  const listComponentArray = allList.map(list => <List key={list.id} header={list.header} cards={list.cardIds.map(id => props.store.allCards[id])}/>)

  return (
    <main className='App'>
      <header className='App-header'>
        <h1>Trelloyes!</h1>
      </header>
      <div className='App-list'>
        {listComponentArray}
      </div>
    </main>
  );
}

export default App;
