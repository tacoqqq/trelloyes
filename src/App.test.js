import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import List from './List';
import renderer from 'react-test-renderer';
import STORE from './store.js'
import App from './App';


describe('Card Component', () => {
    it('renders Card component without error', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer.
            create(<Card id="1" title="First Card" content="Lorem ipsum"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})


describe('List component', () => {

    it('renders component without error', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key="1" header="First List" cards={[{id:"1",title:"First Card",content:"Lorem ipsum"}]} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });


    it('renders the UI as expected', () => {
        const treeOne = renderer.
            create(<List key="2" header="First List" cards={[{id:"1",title:"First Card",content:"Lorem ipsum"}]} />)
            .toJSON();
        expect(treeOne).toMatchSnapshot();
    })
  
})
