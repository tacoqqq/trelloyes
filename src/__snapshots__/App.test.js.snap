// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    First Card
  </h3>
  <p>
    Lorem ipsum
  </p>
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    
  </div>
</section>
`;
